pkg/client/query/codec.go

```diff
@@ -0,0 +1,18 @@
+package query
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// queryCodec is a codec used to unmarshal the account interface returned by the
```

Comment:
1. I feel like this comment is very specific to a specific use case, but can actually be applicable to so much more.
2. Can you add a `TODO_TEHDEBT` to make sure we use this everywhere?

cc @bryanchriswhite

Reply:
I think the idea of using this codec here is that it will only be used here. We only need it when we decode an account from the chain. Thats what this interface provides and should only be used via the interface not the codec.

Previously we had this codec defined in multiple places but now this codec registering this stuff from the auth module allows it to be only used via the interface which is what should be used everyhwere.

WDYT?